fem/libceed/ceed.cpp

```diff
@@ -178,13 +180,16 @@ static void InitCeedNonTensorBasisAndRestriction(const FiniteElementSpace &fes,
          }
       }
    }
-   CeedBasisCreateH1(ceed, GetCeedTopology(fe->GetGeomType()), fes.GetVDim(),
-                     fe->GetDof(), ir.GetNPoints(), shape.GetData(),
-                     grad.GetData(), qref.GetData(), qweight.GetData(), basis);
-   CeedElemRestrictionCreate(ceed, mesh->GetNE(), fe->GetDof(), fes.GetVDim(),
-                             compstride, (fes.GetVDim())*(fes.GetNDofs()),
-                             CEED_MEM_HOST, CEED_COPY_VALUES,
-                             tp_el_dof.GetData(), restr);
+   if (basis)
+      CeedBasisCreateH1(ceed, GetCeedTopology(fe->GetGeomType()), fes.GetVDim(),
+                        fe->GetDof(), ir.GetNPoints(), shape.GetData(),
+                        grad.GetData(), qref.GetData(), qweight.GetData(), basis);
+
+   if (restr)
+      CeedElemRestrictionCreate(ceed, mesh->GetNE(), fe->GetDof(), fes.GetVDim(),
+                                compstride, (fes.GetVDim())*(fes.GetNDofs()),
+                                CEED_MEM_HOST, CEED_COPY_VALUES,
+                                tp_el_dof.GetData(), restr);
```

Comment:
This is really not clean to put the `if` at this level, you created everything to call these functions right before, you shouldn't do the computation if you already know that you're not going to use the result.